exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -173,8 +173,18 @@ public ParquetGroupScan( //
     final FileSelection fileSelection = expandIfNecessary(selection);
 
     this.entries = Lists.newArrayList();
-    for (String fileName : fileSelection.getFiles()) {
-      entries.add(new ReadEntryWithPath(fileName));
+    if (fileSelection.getMetaContext() != null &&
+        (fileSelection.getMetaContext().wasPruningStarted() &&
+        ! fileSelection.getMetaContext().wasPruned())) {
+      // if pruning was attempted and nothing was pruned, initialize the entries with just
+      // the selection root instead of the fully expanded list to reduce overhead. The fully
+      // expanded list is already stored as part of the fileSet.
+      // TODO: at some point we should examine whether the list of entries is absolutely needed.
```

Comment:
Maybe we log a JIRA for this TODO, so that we keep track of this issue? Seems it makes sense to either use the list of entries, or fileSet, but not both.